File: netbox/netbox-snipeit.py
#!/usr/bin/python3
# This file is copyright under the latest version of the EUPL.
# Please see LICENSE file for your rights under this license.
import requests
import cgi
import cgitb
import json

#
# Snipe-IT HOST and API token
#
snipeit_host = 'https://SNIPE-IT_HOST'
token = 'Bearer SNIPE-IT_API_TOKEN'

#
# get form data
#
cgitb.enable()
form = cgi.FieldStorage()

#
# handle devices
#
if "device" in form:
    device = form.getvalue('device')
    url = snipeit_host + '/api/v1/hardware?limit=1&offset=0&search='
    headers = {'authorization': token}
    r = requests.get(url + device, headers = headers)

    #
    # if status code is 200, then redirect to hardware
    #
    if r.status_code == requests.codes.ok:
        #
        # Snipe-IT must return something like an ID
        # If there is no ID, then just list all hardware
        #
        try:
            data = r.json()
            device_id = str(data['rows'][0]['id'])
            print ('Location: ' + snipeit_host + '/hardware/' + device_id + '\n')
        except IndexError:
            print ('Location: ' + snipeit_host + '/hardware/' + '\n')
    else:
        print ('Location: ' + snipeit_host +'\n')
#
# handle locations
#
elif "location" in form:
    location = form.getvalue('location')
    url = snipeit_host + '/api/v1/locations?limit=1&offset=0&search='
    headers = {'authorization': token}
    r = requests.get(url + location, headers = headers)

    #
    # if status code is 200, then redirect to location
    #
    if r.status_code == requests.codes.ok:
        #
        # Snipe-IT must return something like an ID
        # If there is no ID, then just list all locations
        #
        try:
            data = r.json()
            location_id = str(data['rows'][0]['id'])
            print ('Location: ' + snipeit_host + '/locations/' + location_id + '\n')
        except IndexError:
            print ('Location: ' + snipeit_host + '/locations/' +'\n')
    else:
        print ('Location: ' + snipeit_host +'\n')
#
# handle all other errors by just forwarding to the Snipe-IT host
#
else:
    print ('Location: ' + snipeit_host +'\n')

File: netbox/netbox-snipeit.php
<?php
#
# Initialize a snipeit array with token and URL to location API function
#
$snipeit = Array(
	'token' => 'Bearer API_TOKEN_GOES_HERE
	'accept' => 'application/json',
	'content-type' => 'application/json',
	'url' => 'https://SNIPEIT_FQDN/api/v1/locations?limit=1&offset=0&search=',
);
#
# Verify if a location parameter is passed as HTTP GET
#
if(isset($_GET['location'])) {
	#
	# Set CURL options and get data
	#
	$ch = curl_init($snipeit['url'] . $_GET['location']);
	curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
	curl_setopt($ch, CURLOPT_HEADER, 0);
	curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
	curl_setopt($ch, CURLOPT_URL, $snipeit['url'] . $_GET['location']);
	curl_setopt($ch, CURLOPT_HTTPHEADER, Array('authorization: ' . $snipeit['token']));
	$data = curl_exec($ch);
	#
	# If CURL returns HTTP status code 200 (OK), then we can
	# parse the json data and redirect to Snipe-IT location
	#
	if(curl_getinfo($ch, CURLINFO_HTTP_CODE) == 200){
		#
		# use json data as object
		#
		$object = json_decode($data);
			#
			# DEBUG to verify output
			#
			#print_r($object);
			#print $object->rows[0]->id;
		#
		# Do an HTTP redirect to Snipe-IT location by the ID
		#
		$redirect = 'Location: https://SNIPEIT_FQDN/locations/' . $object->rows[0]->id;
		header($redirect);
		exit;
	}
	curl_close($ch);
}
?>